src/Tribe/Integrations/WPML/Meta.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Translate post ids in Event meta data.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Integrations__WPML__Meta {
+
+	/**
+	 * @since TBD
+	 *
+	 * @var Tribe__Events__Integrations__WPML__Meta
+	 */
+	protected static $instance;
+
+	/**
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Integrations__WPML__Meta
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Translates post id in the Event meta data.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return string The translated id for _EventOrganizerID & _EventVenueID.
+	 */
+	public function translate_post_id( $value, $object_id, $meta_key ) {
+		global $wpdb;
+
+		$accepted_values = [ '_EventOrganizerID', '_EventOrganizerID_Order', '_EventVenueID' ];
+
+		if ( ! in_array( $meta_key, $accepted_values ) ) {
+			return $value;
+		}
+
+		$value = $wpdb->get_col( $wpdb->prepare(
```

Comment:
Can we move this to it's own method? To allow for better testing.

Reply:
ready